src/Service/ForwardingConnectionHandler.cpp

```diff
@@ -348,6 +348,12 @@ void ForwardingConnectionHandler::onSocketError(const AutoPtr<ErrorNotification>
 
 void ForwardingConnectionHandler::sendResponse(ForwardResponsePtr response)
 {
+    if (response->forwardType() == ForwardType::Destroy)
+    {
+        delete this;
```

Comment:
I think delete should occur on the IO thread to avoid unexpected behavior, so you should invoke the delete in `onSocketWritable` method.